src/Entities/Aero/AeroBlockSlingshot.cs

```diff
@@ -0,0 +1,417 @@
+    {
+        Idle,
+        Windup,
+        Locked,
+        Launch,
+        Cooldown,
+    }
+
+    public AeroBlockSlingshot(EntityData data, Vector2 offset)
+        : this(data.NodesWithPosition(offset), data.Width, data.Height,
+            data.Float("launchTime", DefaultLaunchTime),
+            data.Float("cooldownTime", DefaultCooldownTime),
+            data.Float("setTime", DefaultSetTime),
+            data.Float("delayTime", DefaultDelayTime),
+            data.Float("pushSpeed", DefaultPushSpeed),
+            data.Enum("pushActions", DefaultPushActions),
+            data.Bool("allowAdjustments", DefaultAllowAdjustments),
+            data.Attr("startColor", DefaultStartColor),
+            data.Attr("endColor", DefaultEndColor))
+    {
+    }
+
+    public AeroBlockSlingshot(Vector2[] positions, int width, int height,
+        float launchTime = DefaultLaunchTime,
+        float cooldownTime = DefaultCooldownTime,
+        float setTime = DefaultSetTime,
+        float delayTime = DefaultDelayTime,
+        float pushSpeed = DefaultPushSpeed,
+        Pushable.MoveActionType pushActions = DefaultPushActions,
+        bool allowAdjustments = DefaultAllowAdjustments,
+        string startColor = DefaultStartColor,
+        string endColor = DefaultEndColor)
+        : base(positions[0], width, height)
+    {
+        LaunchTime = launchTime;
+        CooldownTime = cooldownTime;
+        SetTime = setTime;
+        DelayTime = delayTime;
+        PushSpeed = pushSpeed;
+        AllowAdjustments = allowAdjustments;
+        StartColor = Calc.HexToColor(startColor);
+        EndColor = Calc.HexToColor(endColor);
+
+        startPosition = Position;
+
+        for (int i = 0; i < positions.Length; i++)
+            positions[i].Y = startPosition.Y;
+
+        this.positions = positions;
+
+        sortedPositions = positions.OrderBy(p => p.X).ToArray();
+        leftPosition = sortedPositions.First();
+        rightPosition = sortedPositions.Last();
+
+        Add(pushable = new Pushable
+        {
+            OnPush = OnPush,
+            PushCheck = PushCheck,
+            MaxPushSpeed = PushSpeed,
+            Active = false,
+            MoveActions = pushActions,
+        });
+
+        Add(sfx = new SoundSource()
+        {
+            Position = new Vector2(width, height) / 2.0f,
+        });
+
+        stateMachine = new StateMachine();
+        stateMachine.SetCallbacks((int) SlingshotStates.Idle, IdleUpdate, begin: IdleBegin, end: IdleEnd);
+        stateMachine.SetCallbacks((int) SlingshotStates.Windup, WindupUpdate, begin: WindupBegin, end: WindupEnd);
+        stateMachine.SetCallbacks((int) SlingshotStates.Locked, LockedUpdate, begin: LockedBegin, end: LockedEnd);
+        stateMachine.SetCallbacks((int) SlingshotStates.Launch, LaunchUpdate, begin: LaunchBegin, end: LaunchEnd);
+        stateMachine.SetCallbacks((int) SlingshotStates.Cooldown, CooldownUpdate, begin: CooldownBegin, end: CooldownEnd);
```

Comment:
`StateMachine` has a `IEnumerator` callback for each state, which can be used as a regular coroutine. For states that cannot be interrupted, like Launch and Cooldown, we could use the coroutine callbacks and `yield return Util.Interpolate` like we used to, instead of relying on a timer field

Reply:
I looked at using a per-state coroutine but I just wanted it to be done.  Maybe a future task.